src/searcher/mod.rs

```diff
@@ -349,11 +350,53 @@ impl Searcher {
         Ok(())
     }
 
+    pub fn search_slice_callback_no_path(
```

Comment:
The "organization" of these searcher methods is probably weird but not like I 100% understand them in the first place

Here there was already a `.search_slice_callback()` method (used internally by the `.search_path_callback()` "entry point") but it expected a `Path` to be passed and per above comment its `callback` argument took the additional file contents/path arguments

So just made a separate `.search_slice_callback_no_path()` "entry point"